CMakeLists.txt

```diff
@@ -441,6 +441,18 @@ elseif(MSVC)
                             ${MSVC_LEVEL4_WARNINGS_LIST})
   set(CMAKE_C_FLAGS   "-utf-8 -Wall -WX ${MSVC_DISABLED_WARNINGS_STR} ${MSVC_LEVEL4_WARNINGS_STR}")
   set(CMAKE_CXX_FLAGS "-utf-8 -Wall -WX ${MSVC_DISABLED_WARNINGS_STR} ${MSVC_LEVEL4_WARNINGS_STR}")
+
+  # If we're using MSVC on Windows in FIPS mode with RelWithDebInfo then we want to replace the '-g' debug flag with '/Zi'.
+  # The '/Zi' flag accomplishes the same thing as '-g', but in a manner that is compatible with MSVC.
+  # This also fixes the problem we run into with RelWithDebInfo and FIPS mode on Windows where the FIPS module wouldn't span the expected symbol.
+  if(CMAKE_BUILD_TYPE_LOWER MATCHES "relwithdebinfo" AND FIPS)
+    message(STATUS "Replacing default -g flag for ${CMAKE_BUILD_TYPE} with /Zi")
+    string(REPLACE "-g" "/Zi" CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
+    string(REPLACE "-g" "/Zi" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
+
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} /DEBUG:FULL /OPT:REF,ICF,LBR")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DEBUG:FULL /OPT:REF,ICF,LBR")
```

Comment:
Can you explain what these options are doing.

Reply:
When the `/Zi` compiler flag is set, Visual Studio needs also the linker flag `/DEBUG:FULL` (or `/DEBUG:FASTLINK` or just regular `/DEBUG`) to generate `.pdb` files for any given executable/library. However, `/DEBUG:FULL` by default turns off optimizations, so we need to turn them back on using `/OPT:REF,ICF,LBR`.